modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java

```diff
@@ -44,7 +48,36 @@ public static String getDBType(String dbType, String type) {
 		return dbType;
 	}
 
+	public static long getListTypeDefinitionId(
+		long companyId,
+		ListTypeDefinitionLocalService listTypeDefinitionLocalService,
+		ObjectField objectField) {
+
+		if (!StringUtil.equals(
+				objectField.getBusinessTypeAsString(),
+				ObjectFieldConstants.BUSINESS_TYPE_PICKLIST)) {
+
+			return 0L;
+		}
+
+		long listTypeDefinitionId = GetterUtil.getLong(
+			objectField.getListTypeDefinitionId());
+
+		if (listTypeDefinitionId != 0L) {
+			return objectField.getListTypeDefinitionId();
+		}
+
+		ListTypeDefinition listTypeDefinition =
+			listTypeDefinitionLocalService.
+				fetchListTypeDefinitionByExternalReferenceCode(
+					companyId,
+					objectField.getListTypeDefinitionExternalReferenceCode());
+
+		return listTypeDefinition.getListTypeDefinitionId();
```

Comment:
@murilostodolni as mentioned above, I think we should include the following code here:

```
if (Validator.isNull(objectField.getListTypeDefinitionExternalReferenceCode()) {
   return 0L;
}
```

Reply:
Hi @carolmariaabb in this case we can add this validation in [this if](https://github.com/liferay-objects/liferay-portal/pull/1230/commits/86fa2c5fca1de8193a3d4bb5a4d2345f4ad43fd8#:~:text=ObjectField%20objectField)%20%7B-,if%20(!StringUtil.equals(,%7D,-long%20listTypeDefinitionId%20%3D). What do you think?
```
		if (!StringUtil.equals(
				objectField.getBusinessTypeAsString(),
				ObjectFieldConstants.BUSINESS_TYPE_PICKLIST) ||
                                Validator.isNull(objectField.getListTypeDefinitionExternalReferenceCode()) {

			return 0L;
		}
```